### ISSUE TITLE
Still having issues with Disabling Security hub Controls in LZA (1.7.0)

### ISSUE THREAD

**Issue by frankscalzo** [awslabs/landing-zone-accelerator-on-aws#473]
When adding controls to disabled they never disable the onyl time i have sucess is either manual disable or create a control policy in security hub which than breaks Security hub deployment in LZA with a managed policy error 

Currently the code is
 
      - name: NIST Special Publication 800-53 Revision 5
        # [https://docs.aws.amazon.com/securityhub/latest/userguide/nist-standard.html](https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.aws.amazon.com%2Fsecurityhub%2Flatest%2Fuserguide%2Fnist-standard.html&data=05%7C02%7CFScalzo%40optimumhit.com%7C1c50553fc46744a7345f08dc8643ae7a%7C7853fe62ef514774ab1b876fd9cf540b%7C0%7C0%7C638532872467492325%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=BHd%2BKj55cgzeYdTeKOpR4cQ%2BK3%2F6ZRfJuuQ9ERwBDAs%3D&reserved=0)
        deploymentTargets:
         organizationalUnits:
           -  Root
        enable: true
        controlsToDisable:
          - IAM.6
          - CloudFormation.1
          - Config.1
          - EC2.4
          - EC2.17
          - EC2.8
          - 
And it does not seem to work that being said I wonder if NIST needs to be added to it like PCI has it
 
      - name: PCI DSS v3.2.1
        # [https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-pci-controls.html](https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.aws.amazon.com%2Fsecurityhub%2Flatest%2Fuserguide%2Fsecurityhub-pci-controls.html&data=05%7C02%7CFScalzo%40optimumhit.com%7C1c50553fc46744a7345f08dc8643ae7a%7C7853fe62ef514774ab1b876fd9cf540b%7C0%7C0%7C638532872467502240%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=83QcoRLDg%2FRjUJqX6R9LwUNBtYheUzobTFNhfOfayUo%3D&reserved=0)
        enable: false
        controlsToDisable:
          - PCI.IAM.6
          - PCI.IAM.4
          - PCI.CloudFormation.
I do have an active ticket with AWS so i will report back if they solve

**Comment by akobara** [comment]
You need to specify the controls to disable as they appear in the standard.

For NIST Special Publication 800-53 Revision 5,
Config.1 should be CM-3 CM-6 CM-8

For CIS Foundations Benchmark v1.4.0,
Config.1 is "3.5" (and the quotation marks are required)

You can find those IDs next to the standard in the according user guides:
e.g. https://docs.aws.amazon.com/securityhub/latest/userguide/config-controls.html

**Comment by frankscalzo** [comment]
@akobara seems overly complex i am better off using the policy and not using LZA at this point because if im reading  you right 

to remove cloudtrail.5 i would have to add all these in  
- AC-2(4)
- AC-4(26)
- AC-6(9)
- AU-10
- AU-12
- AU-2
- AU-3
- AU-6(1)
- AU-6(3)
- AU-6(4)
- AU-6(5)
- AU-7(1)
- CA-7
- SC-7(9)
- SI-20
- SI-3(8)
- SI-4(20)
- SI-4(5)
- SI-7(8)

**Comment by yermulnik** [comment]
TL;DR: for `NIST Special Publication 800-53 Revision 5` the `Config.1` is `Config.1`.

**Comment by fdjahnke** [comment]
Hello @frankscalzo, thank you for filing an issue with the Landing Zone Accelerator team!
  
We've confirmed this as a bug and a fix has been merged. It is scheduled to ship in the next release, and we'll follow up here once that release is available.
  
For context: the matching logic for `controlsToDisable` did not account for the control-ID prefixes that some standards use (for example, `PCI.`), so controls listed in those forms were not being disabled. Two notes that may help in the meantime:
  
  - `controlsToDisable` expects AWS Security Hub control IDs (for example, `Config.1`, `IAM.6`, `EC2.8`). Make sure the IDs match what Security Hub returns for the enabled standard.
  - AWS Foundational Security Best Practices control IDs were not affected by this bug; the fix primarily addresses the prefixed IDs.
  
Thanks for supporting Landing Zone Accelerator!

**Comment by fdjahnke** [comment]
Hello @frankscalzo, following up — the fix confirmed earlier has shipped in **[v1.16.0](https://github.com/awslabs/landing-zone-accelerator-on-aws/releases/tag/v1.16.0)**.

The `controlsToDisable` matching logic now normalizes control-ID prefixes on both sides, so controls listed with a standard prefix (for example `PCI.`) are matched and disabled correctly. As noted earlier, `controlsToDisable` expects AWS Security Hub control IDs (for example `Config.1`, `IAM.6`), and AWS Foundational Security Best Practices IDs were not affected by the original bug.

@CastielCloud / @yermulnik — thank you for the NIST 800-53 examples and the finding-JSON walkthrough; using the exact `ControlId` Security Hub returns for the enabled standard (quoted when it's a bare decimal) is the right approach and works with the normalized matching in v1.16.0.

Closing this out as released. Please re-open and ping me if a specific control still won't disable after upgrading. Thank you for supporting Landing Zone Accelerator!
